src/lib/confirm-bar.js

```diff
@@ -35,120 +42,145 @@ chrome.runtime.onMessage.addListener(function({message, iconData, closeData}) {
+          font-family: Lucida Grande, Tahoma, sans-serif;
+          font-size: 14px;
+          text-shadow: none;
+          min-height: 40px;
+          margin: 0;
+          position: fixed;
+          left: 0px;
+          top: -45px;
+          transition: top 300ms ease-in-out;
+          width: 100%;
+          z-index: ${Number.MAX_SAFE_INTEGER};
+        }
+        #${confirmationId}.shown {
+          top: 0;
+        }
+        #${confirmationId} img {
+          height: 24px;
+          width: 24px;
+        }
+        #${confirmationId} button {
+          padding: 0;
+          background-color: #fbfbfb;
+          border: 1px solid #b1b1b1;
+          font-family: Lucida Grande, Tahoma, sans-serif;
+          font-size: 12px;
+          height: 24px;
+          width: 6em;
+        }
+        #${confirmationId} button:hover {
+          background-color: #ebebeb;
+        }
+        #${confirmationId} button:active {
+          background-color: #d4d4d4;
+        }
+        #${confirmationId} button.ok {
+          border: 1px solid #258ad6;
+          background-color: #0ba0f9;
+          color: #fff;
+        }
+        #${confirmationId} button.ok:hover {
+          background-color: #0670cc;
+        }
+        #${confirmationId} button.ok:active {
+          background-color: #005bab;
+        }
+        #${confirmationId} input, #${confirmationId} label {
+          color: #5d5d5d;
+          font-size: 10px;
+        }
+        #${confirmationId} label {
+          margin-left: 3px;
+        }
+        #${confirmationId} .spacer {
+          flex: 1;
+        }
+        #${confirmationId} #${closeId} {
+          cursor: pointer;
+          height: 10px;
+          margin-right: 16px;
+          width: 10px;
+        }
+      </style>
+      <img src="${iconData}" alt="${iconAltText}">
+      <span>${timeTitle}</span>
+      <button class="ok" id="${okId}">${okTitle}</button>
+      <button id="${cancelId}">${cancelTitle}</button>
+      <div class="spacer"></div>
+      <input type="checkbox" id="${showId}"/><label for="${showId}">${dontShowLabel}</label>
+      <img id="${closeId}" src="${closeData}" alt="${closeAltText}">
+    </div>`;
+
+    confirmationBar = iframeDocument.getElementById(confirmationId);
+    const showCheckbox = iframeDocument.getElementById(showId);
+
+    window.setTimeout(() => {
+      confirmationBar.classList.add('shown');
+    }, 500);
+
+    const okButton = iframeDocument.getElementById(okId);
+    okButton.addEventListener('click', () => {
+      browser.runtime.sendMessage({
+        'op': 'confirm',
+        'message': message
+      });
+      confirmationBar.classList.toggle('shown');
+      removeConfirmBar();
+      sendShowUpdate(showCheckbox);
+    });
+
+    const cancelButton = iframeDocument.getElementById(cancelId);
+    cancelButton.addEventListener('click', () => {
+      confirmationBar.classList.toggle('shown');
+      window.setTimeout(removeConfirmBar, 300);
```

Comment:
Why do we need to remove the confirm bar in this code, and not in the previous code?

Reply:
You mean previous as in before the iframe? I think we should have been doing it there, too. The markup hangs around in the DOM that we probably should have cleaned up, even though things were hidden by styles